csi/src/meta_data.rs

```diff
@@ -786,47 +857,36 @@ impl MetaData {
 
                 debug_assert!(
                     dst_path.is_dir(),
-                    "the volume of monnt access type should have a directory path: {:?}",
+                    "the volume of mount access type should have a directory path: {:?}",
                     dst_path,
                 );
-                let open_res = File::open(&src_snapshot.snap_path)
-                    .context(format!("failed to open path={:?}", src_snapshot.snap_path));
-                let tar_gz = match open_res {
-                    Ok(tg) => tg,
-                    Err(e) => {
-                        return Err((RpcStatusCode::INTERNAL, e));
-                    }
-                };
-                let tar_file = flate2::read::GzDecoder::new(tar_gz);
-                let mut archive = tar::Archive::new(tar_file);
-                let unpack_res = archive
-                    .unpack(&dst_path)
-                    .context(format!("failed to decompress snapshot to {:?}", dst_path));
-                if let Err(e) = unpack_res {
-                    return Err((RpcStatusCode::INTERNAL, e));
-                }
+                let snap_path_owned = src_snapshot.snap_path.to_owned();
+                let dst_path_owned = dst_path.to_owned();
+                smol::unblock(move || Self::decompress_snapshot(&snap_path_owned, &dst_path_owned))
+                    .await
+                    .map_err(|e| (RpcStatusCode::INTERNAL, e))?
             }
         }
 
         Ok(())
     }
 
     /// Populate the given destPath with data from the `src_volume_id`
-    pub fn copy_volume_from_volume(
+    pub async fn copy_volume_from_volume(
         &self,
         dst_volume_size: i64,
         src_volume_id: &str,
         dst_volume_id: &str,
     ) -> Result<(), (RpcStatusCode, anyhow::Error)> {
         let dst_path = self.get_volume_path(dst_volume_id);
 
-        match self.get_volume_by_id(src_volume_id) {
+        match self.get_volume_by_id(src_volume_id).await {
             None => {
                 return Err((
                     RpcStatusCode::NOT_FOUND,
                     anyhow!(format!(
                         "failed to find source volume ID={}, \
-                            make sure source/destination in the same storage class",
+                        make sure source/destination in the same storage class",
```

Comment:
extra 4 spaces indentation before the second text line

Reply:
Done